imod/mf6/simulation.py

```diff
@@ -814,6 +814,17 @@ def dump(
                 toml_content[cls_name][key] = model_toml_path.relative_to(
                     directory
                 ).as_posix()
+            elif key in ["gwtgwf_exchanges","split_exchanges"]:
+                toml_content[key] = {}
+                for exchange_package in self[key]:
+                    exchange_type, filename, _, _ = exchange_package.get_specification()
+                    exchange_class_short = type(exchange_package).__name__
+                    if not exchange_class_short in toml_content[key].keys():
+                         toml_content[key][exchange_class_short]=[]
+                    path = f"{filename}.nc"
+                    exchange_package.dataset.to_netcdf(directory / path)
+                    toml_content[key][exchange_class_short].append(path)
```

Comment:
You can reduce this code with a ``defaultdict`` in the native ``collections`` module:

```suggestion
                toml_content[key] = defaultdict(list)
                for exchange_package in self[key]:
                    exchange_type, filename, _, _ = exchange_package.get_specification()
                    exchange_class_short = type(exchange_package).__name__
                    path = f"{filename}.nc"
                    exchange_package.dataset.to_netcdf(directory / path)
                    toml_content[key][exchange_class_short].append(path)
```